### PR TITLE
HHH-11651 - unwrapping error in AbstractMultiTenantConnectionProvider

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-11651

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/spi/AbstractMultiTenantConnectionProvider.java [hunk 1]
```diff
@@ -48,15 +48,17 @@ public boolean supportsAggressiveRelease() {
...
+		return MultiTenantConnectionProvider.class.equals( unwrapType ) ||
 				AbstractMultiTenantConnectionProvider.class.isAssignableFrom( unwrapType );
 	}
 
 	@Override
-	@SuppressWarnings( {"unchecked"})
+	@SuppressWarnings({"unchecked"})
 	public <T> T unwrap(Class<T> unwrapType) {
-		if ( isUnwrappableAs( unwrapType ) ) {
+		if ( ConnectionProvider.class.equals( unwrapType ) ) {
```
Comment (cm-rudolph) [inline hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/spi/AbstractMultiTenantConnectionProvider.java]: The new `AbstractMultiTenantConnectionProvider#isUnwrappableAs` contradicts the ability to unwrap to a `ConnectionProvider`.

File: hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/spi/AbstractMultiTenantConnectionProvider.java [hunk 2]
```diff
@@ -48,15 +48,17 @@ public boolean supportsAggressiveRelease() {
...
+			return (T) getAnyConnectionProvider();
```
Comment (cm-rudolph) [inline hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/spi/AbstractMultiTenantConnectionProvider.java]: Unwrapping to anyConnectionProvider might be unintuitive in some use cases (maybe someone wants to get a tenant specific `ConnectionProvider`), so I would strongly suggest leaving the `unwrap` method as it was before the pull request. The testcase for HHH-11651 would also need to be adjusted to reflect the fact that unwrapping to a `ConnectionProvider` is not supported. `testUnwrappingConnectionProvider` should expect an `UnknownUnwrapTypeException` to be thrown.

**Comment by vladmihalcea** [comment]
We should add a test and address the `AbstractDataSourceBasedMultiTenantConnectionProviderImpl` as well since it might be affected by the same issue.

**Comment by vladmihalcea** [comment]
I submitted a new [Pull Request](https://github.com/hibernate/hibernate-orm/pull/1903) which supersedes this one.

@dreab8 @cm-rudolph Please review the new Pull Request and let me know what you think.

**Pull request closed without merging**